server.R

```diff
@@ -19,118 +19,159 @@ server <- function(input, output, session) {
+            checkboxInput(inputId = "toggle_log",
+                          label = "Log2 scale y-axis",
+                          value = TRUE)
+          )
+        })
+      }
+      else {
+        if(input$subtype == "Mutations") {
+          shinyjs::show(id = "gene")
+          shinyjs::show(id = "mutation_status")
+
+          query <- clinicalQuery(factors = "Gene", table = "mutation", dataset = input$dataset, type = "Gene", unique = TRUE, sort = TRUE)
+          mutation_choices <- dbGetQuery(database, query)
+          updateSelectizeInput(session, "mutation_status", choices = mutation_choices$Gene, selected = NULL, server = TRUE)
+        }
+        else if(input$subtype != "Mutations" && input$subtype != "") {
+          shinyjs::show(id = "gene")
+          shinyjs::show(id = "subtype_options")
+          query <- clinicalQuery(factors=c(input$subtype), type=input$subtype, dataset=input$dataset, unique=TRUE, sort=TRUE)
+          subtype_choices <- unlist(dbGetQuery(database,query),use.names = FALSE)
+          updateCheckboxGroupInput(session,
+                                   "subtype_options",
+                                   label = "Subtypes",
+                                   choices = subtype_choices,
+                                   selected = subtype_choices)
+        }
+        # render median-line checkbox + raw-value checkbox
+        output$toggle_options <- renderUI({
+          tagList(
+            tags$b("Plotting options:"),
+            checkboxInput(inputId = "toggle_median",
+                          label = "Show Median Line",
+                          value = TRUE),
+            checkboxInput(inputId = "toggle_log",
+                          label = "Log2 scale y-axis",
+                          value = TRUE),
+            checkboxInput(inputId = "toggle_boxplot",
+                          label = "Toggle boxplot",
+                          value = FALSE)
+          )
+        })
+      }
     }
   })
 
-  # Reactive value to store the state of the median line
+  # Toggle interactive median line
   show.median <- reactiveVal(TRUE)
-  # Handle button clicks
   observeEvent(input$toggle_median, {
-    # Toggle the value of show.median
     show.median(input$toggle_median)
   })
-  # Reactive value to store the state of the raw value
+
+  # Toggle interactive log2
   show.log2 <- reactiveVal(TRUE)
-  # Handle button clicks
   observeEvent(input$toggle_log, {
-    # Toggle the value of show.log2
     show.log2(input$toggle_log)
   })
 
+  # Toggle interactive boxplot
+  show.boxplot <- reactiveVal(FALSE)
+  observeEvent(input$toggle_boxplot, {
+    show.boxplot(input$toggle_boxplot)
+  })
+
+  # Interactive query data
+  queryData <- reactive({
+    # default value
+    clinical <- NULL
+
+    if (input$subtype == "Multiplot") {
+      query <- geneQuery(genes = input$genes,
+                         table = input$dataset)
+      tcga <- dbGetQuery(database,query)
```

Comment:
let's rename tcga to something else, since it can be beat_aml or some other dataset.  Maybe `expression`?